From aeaceeff667d2af1ab92ed388d6a72ebdd1a423f Mon Sep 17 00:00:00 2001
From: Michael W <baddaemon87@gmail.com>
Date: Mon, 5 Feb 2018 20:10:23 +0100
Subject: [PATCH 099/249] wiki: Better format for release date

* Improve displayed date depending on the given input:
  * "2018" -> 2018
  * "2018-01" -> January 2018
  * "2018-01-01" -> January 01, 2018

Change-Id: I90b256361286f22fd996da140cfeac50ee22ca63
---
 _includes/deviceinfo.html       | 10 ++++++----
 _includes/snippets/release.html |  9 +++++++++
 2 files changed, 15 insertions(+), 4 deletions(-)
 create mode 100644 _includes/snippets/release.html

diff --git a/_includes/deviceinfo.html b/_includes/deviceinfo.html
index 74e7a16..b618b0d 100644
--- a/_includes/deviceinfo.html
+++ b/_includes/deviceinfo.html
@@ -31,15 +31,17 @@ ul, ol {
     <tr>
         <td class="header">Released</td>
         <td>
-            {% if device.release.first -%}
-                {% for model in device.release -%}
-                    {{ model.first[0] }} : {{ model.first[1] }}
+            {%- if device.release.first -%}
+                {%- for model in device.release -%}
+                    {%- assign release = model.first[1] | split: '-' -%}
+                    {{ model.first[0] }}: {% include snippets/release.html -%}
                     {%- unless forloop.last -%}
                     <br/>
                     {% endunless %}
                 {%- endfor %}
             {%- else %}
-                {{ device.release }}
+                {% assign release = device.release | split: '-' -%}
+                {% include snippets/release.html %}
             {%- endif %}
         </td>
     </tr>
diff --git a/_includes/snippets/release.html b/_includes/snippets/release.html
new file mode 100644
index 0000000..b84efd6
--- /dev/null
+++ b/_includes/snippets/release.html
@@ -0,0 +1,9 @@
+{%- assign months = "January,February,March,April,May,June,July,August,September,October,November,December" | split: "," -%}
+{%- assign cur_month = release[1] | plus: 0 -%}
+{%- if release.size == 1 -%}
+{{ release }}
+{%- elsif release.size == 2 -%}
+{{ months[cur_month] }} {{ release[0] }}
+{%- else -%}
+{{ months[cur_month] }} {{ release[2] }}, {{ release[0] }}
+{%- endif -%}
-- 
2.17.1

